### ISSUE TITLE
Changing query size while editing new row to be saved will lose the new row data

### ISSUE THREAD

**Issue by kobaz** [OmniDB/OmniDB#1048]
Bug, as shown here:
![GIFrecord_2019-09-13_162127](https://user-images.githubusercontent.com/1695852/64892539-a707c280-d642-11e9-8a66-721f08c22280.gif)

**Comment by pgaro** [comment]
Is that really a bug? You never saved anything.

Regards
pgaro

**Comment by kobaz** [comment]
I would consider it a bug.  It's data loss.  You're making a view change to show more records and there's no warning about loss of new record input.

I would think it would be useful to either warn the user that they have unsaved data that will be lost, or preserve the data.  My vote would be preserving the data.

**Comment by pgaro** [comment]
I'd vote against that, since there is no way to abort otherwise. I think just discarding what you have done (but not saved) without recalling which row stem from the last action you did, is something helpful for a GUI.

Regards
pgaro

**Comment by wind39** [comment]
Hi @pgaro and @kobaz ,

Thanks for the helpful discussion.

I was thinking that, instead of keeping changes (that may be a lot of changes, imagine you pasted hundreds of thousands of new rows into the grid but didn't click on "Save" yet), we can add a popup that would appear if the user has some non-saved changes and click on the "Play" button to run the query again. The popup would inform the user that there are changes that might be lost, and allow the user to click on "OK" to carry on with the query and lose changes, or click on "Cancel" to get back to the grid.

Regards,

William

**Comment by pgaro** [comment]
Sounds good to me.

Regards
pgaro

**Comment by kobaz** [comment]
Whether or not to keep the changes or show a warning is in the eye of the beholder of course.  I'm a big fan of flexibility...

You can certainly discard unsaved rows after issuing a warning if someone hits the play button.  I would consider this a hard abort on whatever is typed without saving.  But when changing the row count of the query, I would like to see it preserve data.

I believe a casual user would expect their unsaved changes to be preserved as unsaved changes when adjusting the query to add more rows to the result set.  You're not doing anything major, so I don't see the big gain of a warning vs just assuming to preserve the changes.  The user can hard abort if they want by closing the tab.

Basically I did loose data I was entering (and did want the unsaved data preserved) because I was referencing some data from a row above my new row and realized I also needed to see some other rows that were not showing... hence going to the top and adjusting the rows returned... I wasn't sure what to expect, but I did need the additional rows before I did a save, since at the time I was not done with my insert. Nor did I want to save it at that time since the row did not meet minimum insert requirements anyway.  This is the use case for keeping the data and not forcing an 'insert now or else'.
